mesonbuild/options.py

```diff
@@ -529,19 +516,21 @@ class BuiltinOption(T.Generic[_T]):
     """
 
     def __init__(self, opt_type: T.Type[UserOption[_T]], description: str, default: T.Any, yielding: bool = True, *,
-                 choices: T.Any = None, readonly: bool = False):
+                 choices: T.Any = None, readonly: bool = False, **kwargs: object):
         self.opt_type = opt_type
         self.description = description
         self.default = default
         self.choices = choices
         self.yielding = yielding
         self.readonly = readonly
+        self.kwargs = kwargs
 
     def init_option(self, name: 'OptionKey', value: T.Optional[T.Any], prefix: str) -> UserOption[_T]:
         """Create an instance of opt_type and return it."""
         if value is None:
             value = self.prefixed_default(name, prefix)
-        keywords = {'yielding': self.yielding, 'value': value}
+        keywords = {'yielding': self.yielding, 'value_': value}
```

Comment:
Where does that `value_` comes from?

Reply:
Because to use dataclasses I had to rename the argument from `value` to `value_`, otherwise you can't convert the type. See https://github.com/python/mypy/issues/15423, where I argued this was a bug but upstream argued it was the intended behavior.

Reply:
Ah, but python doesn't *need* convertors for dataclasses, do we...

Reply:
I'd laugh if the whole thing didn't make me so mad